subsys/logging/log_core.c

```diff
@@ -184,7 +184,7 @@ static void z_log_msg_post_finalize(void)
 					      K_MSEC(CONFIG_LOG_PROCESS_THREAD_SLEEP_MS),
 					      K_NO_WAIT);
 			} else if (CONFIG_LOG_PROCESS_TRIGGER_THRESHOLD &&
-				   (cnt + 1) == CONFIG_LOG_PROCESS_TRIGGER_THRESHOLD) {
+				   (cnt + 1) >= CONFIG_LOG_PROCESS_TRIGGER_THRESHOLD) {
```

Comment:
This will prolong time needed to log all logging messages when `cnt` exceeds the threshold. What about adding

```
k_timer_stop(&log_process_thread_timer);
k_sem_give(&log_process_thread_sem);
```
to `z_log_dropped` instead? If we start to drop messages then it's a good time to wake up the processing thread.

Can you try if that helps in your case?

Reply:
Confirmed that it also solves my case, changes applied.

Reply:
Hi, is there some action needed from my side for this pull request?

Reply:
No, we are currently in a hard freeze for the v3.7 release.
That is scheduled for release on Friday, and PR's can be merged again after that.